src/Gui/Components/VerticalSplitter.luau

```diff
@@ -0,0 +1,129 @@
+local Hoarcekat = script:FindFirstAncestor("Hoarcekat")
+
+local React = require(Hoarcekat.Packages.React)
+
+local DEFAULT_ALPHA = 0.25
+local HANDLE_WIDTH = 4
+
+local StudioTheme = require(Hoarcekat.Plugin.Gui.Contexts.StudioTheme)
+
+local e = React.createElement
+
+local function VerticalSplitter(props: {
+	mouse: PluginMouse,
+	size: UDim2,
+	left: React.ReactNode,
+	right: React.ReactNode,
+})
+	local theme = React.useContext(StudioTheme.Context)
+
+	local containerRef = React.useRef(nil)
+	local hovering, setHovering = React.useState(false)
+	local dragging, setDragging = React.useState(false)
+	local alpha, setAlpha = React.useState(DEFAULT_ALPHA)
+
+	local onInputBegan = React.useCallback(function(_, input)
```

Comment:
This `VerticalSplitter` component was implemented a short while ago by another person I believe, potentially before Luau's function closure refinements https://luau-lang.org/performance#closure-caching. 

I'll pass the functions directly for the React event instead of using `useCallback`.